cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -198,6 +199,38 @@ runTransactionBuildCmd
         first TxCmdProposalError
           <$> readTxGovernanceActions eon proposalFiles
 
+  -- Extract return addresses from proposals and check that the return address in each proposal is registered
+
+    let returnAddrHashes = Set.fromList --queryStakeAddresses used bellow takes a Set of StakeCredential 
+          [ StakeCredentialByKey returnAddrHash
+          | (proposal, _) <- proposals
+          , let (_, returnAddrHash, _) = fromProposalProcedure eon proposal -- fromProposalProcedure needs to be adjusted so that it works with script hashes. 
+          ]
+    
+    let treasuryWithdrawalAddresses = Set.fromList
+          [ stakeCred
+          | (proposal, _) <- proposals
+          , let (_, _, govAction) = fromProposalProcedure eon proposal
+          , TreasuryWithdrawal withdrawalsList _ <- [govAction]  -- Match on TreasuryWithdrawal action
+          , (_, stakeCred, _) <- withdrawalsList  -- Extract each stake credential in treasury withdrawals
+          ]
+
+    let allAddrHashes = Set.union returnAddrHashes treasuryWithdrawalAddresses
+
+    queryResult <- liftIO $ executeLocalStateQueryExpr localNodeConnInfo Consensus.VolatileTip $
+      queryStakeAddresses eon allAddrHashes networkId
+    
+    let unregisteredAddresses = case queryResult of
+          Right (Right (Right (balances, _))) ->
+            filter (\stakeCred -> not (Map.member (makeStakeAddress networkId stakeCred) balances)) (Set.toList allAddrHashes)
+          _ -> Set.toList allAddrHashes -- If query failed, add to the unregistered addresses list. 
+
+    -- WIP: This is a temporary solution to handle error message.
+    when (not (null unregisteredAddresses)) $ liftIO $ do
+      let errMsg = "Error: One or more stake addresses in proposals is not registered: " ++ show unregisteredAddresses
+      IO.hPutStrLn IO.stderr errMsg
+      throwIO $ userError errMsg 
```

Comment:
Since we are in the `ExceptT` monad, we have a error handling mechanism available and we should use it instead of throwing an unhandled exception like `throwIO` does.